AppLovinMAXDemo/Plugins/AppLovinMAX/Source/AppLovinMAX/AppLovinMAX_UPL_Android.xml

```diff
@@ -0,0 +1,161 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--AppLovin plugin additions-->
+<root xmlns:android="http://schemas.android.com/apk/res/android">
+  <!-- init section is always evaluated once per architecture -->
+  <init>
+    <log text="AppLovin Android SDK init"/>
+
+    <!-- check if using Gradle -->
+    <setBoolFromProperty result="bGradle" ini="Engine" section="/Script/AndroidRuntimeSettings.AndroidRuntimeSettings" property="bEnableGradle" default="false"/>
+
+    <!-- get packaging for Applovin from ini and reset it if architecture not supported -->
+    <setBoolFromProperty result="bEnabled" ini="Engine" section="AppLovin" property="bEnabled" default="true"/>
+    <setStringFromProperty result="ClientId" ini="Engine" section="AppLovin" property="ClientId" default="0"/>
+
+    <dumpvars/>
+  </init>
+
+  <!-- optional files or directories to copy or delete from Intermediate/Android/APK after ndk-build -->
+  <resourceCopies>
+    <copyFile src="$S(PluginDir)/../ThirdParty/Android/applovin-max-unreal-plugin.aar" dst="$S(BuildDir)/libs/applovin-max-unreal-plugin.aar" />
+    <copyFile src="$S(PluginDir)/../ThirdParty/Android/applovin-sdk-10.3.0.aar" dst="$S(BuildDir)/libs/applovin-sdk-10.3.0.aar" />
+  </resourceCopies>
+
+  <!-- optional updates applied to AndroidManifest.xml -->
+  <androidManifestUpdates>
+    <if condition="bEnabled">
+      <true>
+        <addPermission android:name="android.permission.INTERNET" />
+        <addPermission android:name="android.permission.ACCESS_WIFI_STATE" />
+      </true>
+    </if>
+  </androidManifestUpdates>
+
+  <!-- optional additions to proguard -->
+  <proguardAdditions> </proguardAdditions>
+
+  <!-- optional AAR imports additions -->
+  <!-- <AARImports> -->
+  <!--   <if condition="bEnabled"> -->
+  <!--     <true> -->
+  <!--       <insertValue value="com.applovin,applovin-sdk,+@aar" /> -->
+  <!--       <insertNewline/> -->
+  <!--     </true> -->
+  <!--   </if> -->
+  <!-- </AARImports> -->
+
+  <!-- optional base build.gradle additions -->
+  <baseBuildGradleAdditions>
+  </baseBuildGradleAdditions>
+
+  <!-- optional base build.gradle buildscript additions -->
+  <buildscriptGradleAdditions>  </buildscriptGradleAdditions>
+
+  <!-- optional app build.gradle additions -->
+  <buildGradleAdditions>
+    <if condition="bEnabled">
+      <true>
+        <insert>
+          allprojects {
+            repositories {
+              flatDir {
+        </insert>
+        <insertValue value="dirs '$S(BuildDir)/libs'"/>
+        <insertNewline/>
+        <insert>
+              }
+            }
+          }
+
+          dependencies.implementation(name: 'applovin-max-unreal-plugin', ext: 'aar')
+          dependencies.implementation(name: 'applovin-sdk-10.3.0', ext: 'aar')
```

Comment:
We should rely on Gradle to pull in the latest AppLovin SDK automatically, so we don't have to hardcode/handle a specific version

```suggestion
          dependencies.implementation(name: 'com.applovin:applovin-sdk:+', ext: 'aar')
```

Reply:
Yup, that is ideal, but I wonder if Unreal has a limitation or something... Like the SDK needing to be explicitly copied via its [filename](https://github.com/AppLovin/AppLovin-MAX-Unreal/pull/1/files?file-filters%5B%5D=.aar&file-filters%5B%5D=.bat&file-filters%5B%5D=.bin&file-filters%5B%5D=.code-workspace&file-filters%5B%5D=.cpp&file-filters%5B%5D=.cs&file-filters%5B%5D=.dylib&file-filters%5B%5D=.gradle&file-filters%5B%5D=.h&file-filters%5B%5D=.ini&file-filters%5B%5D=.jar&file-filters%5B%5D=.java&file-filters%5B%5D=.json&file-filters%5B%5D=.kts&file-filters%5B%5D=.md&file-filters%5B%5D=.mm&file-filters%5B%5D=.modules&file-filters%5B%5D=.pbxproj&file-filters%5B%5D=.plist&file-filters%5B%5D=.png&file-filters%5B%5D=.pro&file-filters%5B%5D=.properties&file-filters%5B%5D=.uasset&file-filters%5B%5D=.umap&file-filters%5B%5D=.uplugin&file-filters%5B%5D=.uproject&file-filters%5B%5D=.xcscheme&file-filters%5B%5D=.xml&file-filters%5B%5D=.zip&file-filters%5B%5D=No+extension&file-filters%5B%5D=dotfile#diff-5f34f680806f4919ccb8e23c286c50a195273a7325fabceec0bda858a42b1807R21).

Reply:
We can thankfully add the Android SDK using Gradle without needed to copy via filename, specifically using the `<AARImports>` tag in this this xml file.